miden/tests/integration/stdlib/collections/mmr.rs

```diff
@@ -0,0 +1,168 @@
+
+    let test = build_test!(one, &[]);
+    test.expect_stack(&[0, 1 << 0]);
+}
+
+#[test]
+fn test_mmr_get_single_peak() -> Result<(), MerkleError> {
+    // This test uses a single merkle tree as the only MMR peak
+    let leaves = &[1, 2, 3, 4];
+    let mut merkle_store = MerkleStore::new();
+    let merkle_root = merkle_store.add_merkle_tree(init_merkle_leaves(leaves))?;
+    let advice_stack: Vec<u64> = merkle_root.iter().map(StarkField::as_int).collect();
+
+    for pos in 0..(leaves.len() as u64) {
+        let source = format!(
+            "use.std::collections::mmr
+
+            begin
+                push.{num_leaves} push.1000 mem_store # leaves count
+                adv_push.4 push.1001 mem_storew dropw # MMR single peak
+
+                push.1000 push.{pos} exec.mmr::get
+            end",
+            num_leaves = leaves.len(),
+            pos = pos,
+        );
+
+        let test = build_debug_test!(source, &[], advice_stack, merkle_store.clone());
+        let leaf = merkle_store.get_node(merkle_root, NodeIndex::new(2, pos))?;
+
+        // the stack should be first the leaf element followed by the tree root
+        let stack: Vec<u64> = leaf
+            .iter()
+            .map(StarkField::as_int)
+            .rev()
+            .chain(merkle_root.iter().map(StarkField::as_int).rev())
+            .collect();
+        test.expect_stack(&stack);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_mmr_get_two_peaks() -> Result<(), MerkleError> {
+    // This test uses two merkle trees for the MMR, one with 8 elements, and one with 2
+    let leaves1 = &[1, 2, 3, 4, 5, 6, 7, 8];
+    let leaves2 = &[9, 10];
+    let num_leaves = leaves1.len() + leaves2.len();
+
+    let mut merkle_store = MerkleStore::new();
+    let merkle_root1 = merkle_store.add_merkle_tree(init_merkle_leaves(leaves1))?;
+    let merkle_root2 = merkle_store.add_merkle_tree(init_merkle_leaves(leaves2))?;
+
+    let merkle_tree1 = MerkleTree::new(init_merkle_leaves(leaves1))?;
+    assert_eq!(merkle_tree1.depth(), 3);
+    assert_eq!(merkle_tree1.root(), merkle_root1);
+
+    let advice_stack: Vec<u64> = merkle_root1
+        .iter()
+        .map(StarkField::as_int)
+        .chain(merkle_root2.iter().map(StarkField::as_int))
+        .collect();
+
+    let examples = [
+        // root, absolute_pos, leaf
+        (merkle_root1, 0, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 0u64))?),
+        (merkle_root1, 1, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 1u64))?),
+        (merkle_root1, 2, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 2u64))?),
+        (merkle_root1, 3, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 3u64))?),
+        (merkle_root1, 7, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 7u64))?),
+        (merkle_root2, 8, merkle_store.get_node(merkle_root2, NodeIndex::new(1u8, 0u64))?),
+        (merkle_root2, 9, merkle_store.get_node(merkle_root2, NodeIndex::new(1u8, 1u64))?),
+    ];
```

Comment:
note: i have a love/hate relationship with loop in tests. It is super annoying when anything but the first iteration fails (because getting to the breakpoint in the debugger gets a tiny little bit more annoying).

This list was just a hack I did so I could comment out test cases and run that on my debugging sessions. There are no other reasons why I haven't iterated over every leaf. Let me know if that should be changed.

Reply:
I think this is fine.